diff -up syntastic/syntax_checkers/d/dmd.vim.fix syntastic/syntax_checkers/d/dmd.vim
--- syntastic/syntax_checkers/d/dmd.vim.fix	2013-05-19 18:21:43.830190370 +0200
+++ syntastic/syntax_checkers/d/dmd.vim	2013-05-19 18:25:07.487063626 +0200
@@ -83,7 +83,13 @@ endif
 let g:loaded_syntastic_d_dmd_checker = 1
 
 if !exists('g:syntastic_d_compiler')
-    let g:syntastic_d_compiler = 'dmd'
+    if executable('dmd')
+        let g:syntastic_d_compiler = 'dmd'
+    elseif executable('ldmd2')
+        let g:syntastic_d_compiler = 'ldmd2'
+    elseif executable('gdmd')
+        let g:syntastic_d_compiler = 'gdmd'
+    endif
 endif
 
 function! SyntaxCheckers_d_dmd_IsAvailable()
